docs/how-to/unified_memory.rst

```diff
@@ -0,0 +1,595 @@
+
+How-to use?
+===========
+
+Unified Memory Management (UMM) is a feature that can simplify the complexities
+of memory management in GPU computing. It is particularly useful in
+heterogeneous computing environments with heavy memory usage with both a CPU
+and a GPU which would require large memory transfers. Here are some areas where
+UMM can be beneficial:
+
+- **Simplification of Memory Management**:
+UMM can help to simplify the complexities of memory management. This can make
+it easier for developers to write code without having to worry about the
+details of memory allocation and deallocation.
+
+- **Data Migration**:
+UMM allows for efficient data migration between the host (CPU) and the device
+(GPU). This can be particularly useful for applications that need to move data
+back and forth between the device and host.
+
+- **Improved Programming Productivity**:
+As a positive side effect, the use of UMM can reduce the lines of code,
+thereby improving programming productivity.
+
+In HIP, pinned memory allocations are coherent by default. Pinned memory is
+host memory that is mapped into the address space of all GPUs, meaning that the
+pointer can be used on both host and device. Using pinned memory instead of
+pageable memory on the host can lead an improvement in bandwidth.
+
+While UMM can provide numerous benefits, it is also important
+to be aware of the potential performance overhead associated with UMM.
+Therefore, it is recommended to thoroughly test and profile your code to
+ensure it is indeed the most suitable choice for your specific use case.
+
+System Requirements
+===================
+Unified memory is supported on Linux by all modern AMD GPUs from the Vega
+series onward. Unified memory management can be achieved with managed memory
+allocation and, for the latest GPUs, with a system allocator.
+
+The table below lists the supported allocators. The allocators are described in
+the next chapter.
+
+.. list-table:: Supported Unified Memory Allocators
+    :widths: 40, 25, 25, 25
+    :header-rows: 1
+    :align: center
+
+    * - Architecture
+      - ``hipMallocManaged()``
+      - ``__managed__``
+      - ``malloc()``
+    * - MI200, MI300 Series
+      - ✅
+      - ✅
+      - ✅ :sup:`1`
+    * - MI100
+      - ✅
+      - ✅
+      - ❌
+    * - RDNA (Navi) Series
+      - ✅
+      - ✅
+      - ❌
+    * - GCN5 (Vega) Series
+      - ✅
+      - ✅
+      - ❌
+
+✅: **Supported**
+
+❌: **Unsupported**
+
+:sup:`1` Works only with ``XNACK=1``. First GPU access causes recoverable page-fault.
+
+Unified Memory Programming Models
+=================================
+
+- **HIP Managed Memory Allocation API**:
+The ``hipMallocManaged()`` is a dynamic memory allocator available at all GPUs
+with unified memory support. For more details please visit
+:doc:`reference page:reference/unified_memory_reference`.
+
+- **HIP Managed Variables**:
+The ``__managed__`` declaration specifier, which serves as its counterpart, is
```

Comment:
Recommended

The __managed__ declaration specifier, which serves as its counterpart, is supported on all modern AMD cards and can be utilized for static allocation.